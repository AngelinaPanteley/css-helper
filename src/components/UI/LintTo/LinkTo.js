import React from 'react';
import { Link } from 'react-router-dom';
import styles from './LinkTo.scss';

const link = (props) => {
  return (
    <Link to={props.route} className={styles.Link}>
      {props.children}
    </Link>
  )
}

export default link;