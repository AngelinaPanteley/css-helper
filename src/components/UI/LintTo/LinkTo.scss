@import '../../../variables';

.Link {
  text-decoration: none;
  color: $color-white;
  transition: $transition;
  font-size: 30px;
  text-transform: uppercase;
  padding: 20px 40px 15px;
  border: 1px solid #ffffff;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  z-index: 2;

  &:hover,
  &:focus,
  &:active {
    color: $color-light-pink;
    border: 1px solid $color-light-pink;
    transition: $transition;
  }

  @include mobile {
    font-size: 20px;
    line-height: 30px;
  }
}