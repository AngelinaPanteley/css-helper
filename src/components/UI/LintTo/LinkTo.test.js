"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import { expect as chaiExpect } from 'chai';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16.1';
import LinkTo from './LinkTo.js';
import { MemoryRouter, Link } from "react-router-dom";
configure({ adapter: new Adapter() });

describe('>>>LinkTo', () => {

  //SNAPSHOTS

  const props = {
    route: '/route',
    onClick: () => { },
  }

  const linkComponent = <MemoryRouter><LinkTo {...props} /></MemoryRouter>;

  const component = renderer.create(linkComponent);

  it('renders itself correctly', () => {
    let componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();
  });

  //SHALLOW RENDERING

  let wrapper;
  const shallowComponent = <LinkTo {...props} />;

  beforeEach(() => {
    wrapper = shallow(shallowComponent);
  });

  it('renders Link', () => {
    chaiExpect(wrapper.find(Link)).to.have.lengthOf(1);
  });

})