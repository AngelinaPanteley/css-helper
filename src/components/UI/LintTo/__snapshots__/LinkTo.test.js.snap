// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>LinkTo renders itself correctly 1`] = `
<a
  href="/route"
  onClick={[Function]}
/>
`;
